From e4e09ddecf0d514b06c4d0b395260b58140f8ac5 Mon Sep 17 00:00:00 2001
From: David Kershner <dkershner@habana.ai>
Date: Fri, 30 May 2025 23:46:40 +0300
Subject: [PATCH] [SW-210720] boot_info: store per hbm ecc thershold

Save the hbm threshold information in boot info so it can be read by
preboot during the next boot.

Change-Id: I3dc3437612da4feace927b40f70e3f4f7ac3342d
Signed-off-by: David Kershner <dkershner@habana.ai>
---

diff --git a/hl_fw_if.h b/hl_fw_if.h
index 83ebab3..b406eec 100644
--- a/hl_fw_if.h
+++ b/hl_fw_if.h
@@ -144,6 +144,7 @@
 #define HL_NVME_MAX_ECC_SIZE	64
 #define HL_SRAM_ARRAYS_COUNT	3
 #define HL_HBM_ARRAYS_COUNT	3
+#define HL_NUM_HBM_COUNT	8
 
 #define HL_NVME_MAX_PARITY_SIZE	64
 #define HL_PARITY_QUEUE_SIZE	(sizeof(uint32_t) + HL_NVME_MAX_PARITY_SIZE * sizeof(uint16_t))
@@ -185,6 +186,7 @@
 	uint16_t hbm_serr_bursts;
 	uint16_t hbm_fatal_parity;
 	struct hl_mem_hbm_bank_replacement_counters bank_replacement_counters;
+	uint8_t hbm_ecc_thresholds[HL_NUM_HBM_COUNT];
 };
 
 #define HL_VUART_MAGIC_VER(v)	(0x86828400 | (v)) /* VRTx */
